addons/stock/models/procurement.py

```diff
@@ -229,7 +229,7 @@ def _merge_domain(self, values, rule, group_id):
 
     @api.model
     def _get_exceptions_domain(self):
-        return [('procure_method', '=', 'make_to_order'), ('move_orig_ids', '=', False)]
+        return [('procure_method', '=', 'make_to_order'), ('move_orig_ids', '=', False), ('state','!=', 'cancel')]
```

Comment:
What do you think of ('state', 'not in', ('cancel', 'done', 'draft')) instead? In order to avoid other problem later